icu4c/source/i18n/messageformat2_parser.cpp

```diff
@@ -173,18 +173,27 @@ UnicodeSet* initAlpha(UErrorCode& status) {
 UnicodeSet* initDigits(UErrorCode& status) {
     UnicodeSet* result = new UnicodeSet(UnicodeString("[:number:]"), status);
     if (U_FAILURE(status)) {
-        return {};
+        return nullptr;
     }
     result->freeze();
     return result;
 }
 
 UnicodeSet* initNameStartChars(UErrorCode& status) {
     if (U_FAILURE(status)) {
-        return {};
+        return nullptr;
     }
 
-    UnicodeSet* result = new UnicodeSet(*unisets::getImpl(unisets::ALPHA));
+    initAlpha(status);
```

Comment:
initAlpha and other initX return a UnicodeSet*, you just leak the returned result here.

Reply:
You're right; fixed.